enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultControllerEnricher.java

```diff
@@ -125,8 +131,52 @@ public void addMissingResources(KubernetesListBuilder builder) {
                     builder.addToJobItems(jobHandler.getJob(config, images));
                 }
             }
+        } else if (KubernetesResourceUtil.checkForKind(builder, "StatefulSet")) {
+            final StatefulSetSpec spec = statefulSetHandler.getStatefulSet(config, images).getSpec();
+            if (spec != null) {
+                PodTemplateSpec template = spec.getTemplate();
+                if (template != null) {
+                    final PodSpec podSpec = template.getSpec();
+                    if (podSpec != null) {
+                        builder.accept(new TypedVisitor<PodSpecBuilder>() {
+                            @Override
+                            public void visit(PodSpecBuilder builder) {
+                                KubernetesResourceUtil.mergePodSpec(builder, podSpec, name);
+                            }
+                        });
+
+                        // handle StatefulSet YAML which may not have a StatefulSetSpec, PodTemplateSpec or PodSpec
+                        // or if it does lets enrich with the defaults
+                        builder.accept(new TypedVisitor<StatefulSetBuilder>() {
+                            @Override
+                            public void visit(StatefulSetBuilder builder) {
+                                StatefulSetSpec statefulSetSpec = builder.getSpec();
+                                if (statefulSetSpec == null) {
+                                    builder.withNewSpec().endSpec();
+                                    statefulSetSpec = builder.getSpec();
+                                }
+                                mergeStatefulSetSpec(builder, spec);
+                                PodTemplateSpec template = statefulSetSpec.getTemplate();
+                                StatefulSetFluent.SpecNested<StatefulSetBuilder> specBuilder = builder.editSpec();
+                                if (template == null) {
+                                    specBuilder.withNewTemplate().withNewSpecLike(podSpec).endSpec().endTemplate().endSpec();
```

Comment:
I think this can be done more elegantly with `specBuilder.editOrNewTemplateLike(...)` . But @iocanel knows best, really not _fluent_ with the fluent API yet :)

Would probably save some "ifs"